Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -1025,6 +1025,22 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     return .visitChildren
   }
 
+  public override func visit(_ node: SwitchExprSyntax) -> SyntaxVisitorContinueKind {
+    if shouldSkip(node) {
+      return .skipChildren
+    }
+
+    if node.expression.is(MissingExprSyntax.self) && !node.cases.isEmpty {
+      addDiagnostic(node.expression, .missingExpressionInSwitchStatement, handledNodes: [node.expression.id])
+    }
+
+    if node.cases.isEmpty && !node.leftBrace.isMissingAllTokens && !node.rightBrace.isMissingAllTokens {
+      addDiagnostic(node.cases, .emptySwitchExprBody, handledNodes: [node.cases.id])
+    }
```

Comment:
This only works if the `SwiftExprSyntax` has another error because otherwise the `hasError` bit is not set on the `SwitchExprSyntax` and thus we’re skipping the node. You can verify that we are not seeing this issue if the expression is not missing in the `switch`. So, we should consider this diagnostic semantic and it should be diagnosed at a later stage since it doesn’t prevent us from producing a well-formed SwiftSyntax tree.

That is, it’s OK if they are not emitted by SwiftParser even if they were by the old parser.

Reply:
In that case we would still skip the parent of the `SwitchExprSyntax`. It’s just not something we can represent in `ParserDiagnosticGenerator` and it will need to be diagnostic at a later stage.

Reply:
Okay, will remove and add the todo again.
I still think it is something we should handle somewhere 😁

Reply:
You can also remove the TODO. I have removed many of those that we consider semantic. Just missed this one.